packages/react-dom/src/__tests__/ReactDOM-test.js

```diff
@@ -338,4 +338,45 @@ describe('ReactDOM', () => {
     ];
     expect(actual).toEqual(expected);
   });
+
+  it('should call the setState callback even if shouldComponentUpdate = false', () => {
+    const mockFn = jest.fn().mockReturnValue(false);
+    let setState, getState;
+
+    const div = document.createElement('div');
+    document.body.appendChild(div);
+
+    class Component extends React.Component {
+      constructor(props, context) {
+        super(props, context);
+        this.state = {
+          hasUpdatedState: false,
+        };
+      }
+
+      componentWillMount() {
+        setState = (newState, callback) => this.setState(newState, callback);
+        getState = () => this.state;
```

Comment:
These helpers make it a bit hard to read what's really going on in the test. Could you please just store the instance instead, and then read `state` and call `setState` on it directly?